### ISSUE TITLE
Problems with restarting existing maps with latest 0.17.36

### ISSUE THREAD

**Issue by bplein** [dtandersen/docker_factorio_server#250]
The "find" statements in the docker-entrypoint.sh are often "missing" files. So it doesn't find a map named my-map-name.zip and it creates a new empty map. Or it doesn't find the map-settings.json (which exists) and then it creates a new one.

In my cases, this means there are missing sections in the map-settings.json and the map fails to load. Or I get _austosave1.zip when restarting (docker stop, docker rm, then restart with valid save files). It's also failing to clean up *tmp.zip save files.

**Comment by Fank** [comment]
can you please provide the command how you start the container?
And a `tree` of the folder you mount.
This would help us to identify the issue.

**Comment by bplein** [comment]
Here is my template for docker-compose.yml. I read it in, substituting variables such as the version of image, PWD, etc. 
```
$ cat docker-compose-template.yml
version: '2.2'
services:
  factorio:
    image: "factoriotools/factorio:${LATEST}"
    cpus: 1
    cpuset: $MYCPU
    mem_limit: 6G
    mem_reservation: 4G
    restart: always
    network_mode: bridge
    ports:
     - "${GAME_PORT}:${GAME_PORT}/udp"
     - "${RCON_PORT}:${RCON_PORT}/tcp"
    environment:
      - PORT=${GAME_PORT}
      - RCON_PORT=${RCON_PORT}
    container_name: "${CONTAINER_NAME}"
    volumes:
     - ./volume:/factorio:delegated
     - ${PWD}/volume/config/config.ini:/opt/factorio/config/config.ini:delegated
    stdin_open: true
    tty: true
```

And here is the tree for the volume. Note that I have gotten what seems to be inconsistent results, but it seems that simple names without hyphens or extra periods/dots work better, where before it didn't matter, I could have new-oarc.17.34.zip etc.

```

bill@mothra:~/docker/factorio/factorio1/volume$ tree > /tmp/foo.txt
bill@mothra:~/docker/factorio/factorio1/volume$ cat /tmp/foo.txt
.
├── config
│   ├── config.ini
│   ├── map-gen-settings.json
│   ├── map-gen-settings.json.bak
│   ├── map-settings.json
│   ├── rconpw
│   ├── server-adminlist.json
│   ├── server-banlist.json
│   ├── server-id.json
│   ├── server-settings.json
│   └── server-settings.json.template
├── mods
│   └── mod-list.json
├── saves
│   ├── 2018-05-30.zip
│   ├── 2018-07-16.zip
│   ├── 2018-12-26-Railworld.zip
│   ├── _autosave20.zip
│   ├── _autosave21.zip
│   ├── _autosave22.zip
│   ├── _autosave23.zip
│   ├── _autosave24.zip
│   ├── _autosave25.zip
│   ├── _autosave26.zip
│   ├── _autosave27.zip
│   ├── _autosave28.zip
│   ├── _autosave29.zip
│   ├── _autosave2.zip
│   ├── _autosave30.zip
│   ├── _autosave31.zip
│   ├── _autosave32.zip
│   ├── _autosave33.zip
│   ├── _autosave34.zip
│   ├── _autosave35.zip
│   ├── _autosave36.zip
│   ├── _autosave37.zip
│   ├── _autosave38.zip
│   ├── _autosave39.zip
│   ├── _autosave3.zip
│   ├── _autosave40.zip
│   ├── _autosave41.zip
│   ├── _autosave42.zip
│   ├── _autosave43.zip
│   ├── _autosave44.zip
│   ├── _autosave45.zip
│   ├── _autosave46.zip
│   ├── _autosave47.zip
│   ├── _autosave48.zip
│   ├── _autosave49.zip
│   ├── _autosave4.zip
│   ├── _autosave50.zip
│   ├── _autosave51.zip
│   ├── _autosave52.zip
│   ├── _autosave53.zip
│   ├── _autosave54.zip
│   ├── _autosave55.zip
│   ├── _autosave56.zip
│   ├── _autosave57.zip
│   ├── _autosave58.zip
│   ├── _autosave59.zip
│   ├── _autosave5.zip
│   ├── _autosave60.zip
│   ├── _autosave61.zip
│   ├── _autosave62.zip
│   ├── _autosave63.zip
│   ├── _autosave64.zip
│   ├── _autosave65.zip
│   ├── _autosave66.zip
│   ├── _autosave67.zip
│   ├── _autosave68.zip
│   ├── _autosave69.zip
│   ├── _autosave6.zip
│   ├── _autosave70.zip
│   ├── _autosave71.zip
│   ├── _autosave72.zip
│   ├── _autosave73.zip
│   ├── _autosave74.zip
│   ├── _autosave75.zip
│   ├── _autosave76.zip
│   ├── _autosave77.zip
│   ├── _autosave78.zip
│   ├── _autosave79.zip
│   ├── _autosave7.zip
│   ├── _autosave80.zip
│   ├── _autosave81.zip
│   ├── _autosave82.zip
│   ├── _autosave83.zip
│   ├── _autosave84.zip
│   ├── _autosave85.zip
│   ├── _autosave86.zip
│   ├── _autosave87.zip
│   ├── _autosave88.zip
│   ├── _autosave89.zip
│   ├── _autosave8.zip
│   ├── _autosave90.zip
│   ├── _autosave91.zip
│   ├── _autosave92.zip
│   ├── _autosave93.zip
│   ├── _autosave94.zip
│   ├── _autosave95.zip
│   ├── _autosave96.zip
│   ├── _autosave97.zip
│   ├── _autosave98.zip
│   ├── _autosave99.zip
│   ├── _autosave9.zip
│   ├── index.html
│   ├── railworld.zip
│   ├── SquidRail-0.17.2-2018-02-28.zip
│   ├── SquidRail.zip
│   ├── SuperRail.zip
│   └── Wasteland.zip
├── scenarios
│   ├── CrashSite
│   │   └── locale
│   │       ├── en
│   │       ├── softmod-modules-dz
│   │       ├── softmod-modules-stdlib
│   │       ├── softmod-modules-util
│   │       └── utils
│   ├── Grid
│   │   ├── announcements.lua
│   │   ├── belt_limit.lua
│   │   ├── blueprint.dat
│   │   ├── control.lua
│   │   ├── debug.lua
│   │   ├── depreciated
│   │   │   ├── nuclear.lua
│   │   │   ├── rocket.lua
│   │   │   ├── stats.lua
│   │   │   ├── void.lua
│   │   │   └── wave_defense.lua
│   │   ├── equipment.lua
│   │   ├── gravemarker.lua
│   │   ├── info.json
│   │   ├── lib
│   │   │   ├── event.lua
│   │   │   ├── is.lua
│   │   │   └── table.lua
│   │   ├── locale
│   │   │   ├── frontier_silo.lua
│   │   │   ├── oarc_utils.lua
│   │   │   ├── README.md
│   │   │   ├── resources
│   │   │   │   └── poke_messages.lua
│   │   │   ├── rso
│   │   │   │   ├── drand.lua
│   │   │   │   └── metaball.lua
│   │   │   └── tag.lua
│   │   ├── lualib
│   │   │   ├── antigrief.lua
│   │   │   ├── bot.lua
│   │   │   ├── char_mod.lua
│   │   │   ├── event.lua
│   │   │   ├── modular_admin
│   │   │   │   ├── modular_admin_alert.lua
│   │   │   │   ├── modular_admin_boost.lua
│   │   │   │   ├── modular_admin_compensate.lua
│   │   │   │   ├── modular_admin_ghosts.lua
│   │   │   │   ├── modular_admin.lua
│   │   │   │   ├── modular_admin_players.lua
│   │   │   │   ├── modular_admin_spectate.lua
│   │   │   │   └── modular_admin_tag.lua
│   │   │   ├── modular_information
│   │   │   │   ├── modular_information_about.lua
│   │   │   │   ├── modular_information_dummy.lua
│   │   │   │   ├── modular_information.lua
│   │   │   │   ├── modular_information_popup.lua
│   │   │   │   ├── modular_information_rules.lua
│   │   │   │   ├── modular_information_scenario.lua
│   │   │   │   ├── modular_information_stats.lua
│   │   │   │   └── modular_information_team.lua
│   │   │   ├── modular_tag
│   │   │   │   ├── modular_tag.lua
│   │   │   │   └── modular_tag_patreon.lua
│   │   │   ├── pdnc.lua
│   │   │   └── topgui.lua
│   │   ├── maps
│   │   │   ├── dangOreus.lua
│   │   │   ├── grid.lua
│   │   │   ├── heximaze.lua
│   │   │   ├── heximazing.lua
│   │   │   ├── locotrains.lua
│   │   │   ├── perlinvoid.lua
│   │   │   ├── prospector.lua
│   │   │   ├── searious.lua
│   │   │   └── TTSFN.lua
│   │   ├── modules
│   │   │   ├── autofill.lua
│   │   │   ├── bluebuild.lua
│   │   │   ├── bpmirror.lua
│   │   │   ├── dark harvest.lua
│   │   │   ├── dirtpath.lua
│   │   │   ├── divOresity.lua
│   │   │   ├── enhancedbiters.lua
│   │   │   ├── gravemarker.lua
│   │   │   ├── infinity.lua
│   │   │   ├── lazybastard.lua
│   │   │   ├── market.lua
│   │   │   ├── masspower.lua
│   │   │   ├── mesh.lua
│   │   │   ├── module_list.lua
│   │   │   ├── nougatmining.lua
│   │   │   ├── nyan.lua
│   │   │   ├── peppermintmining.lua
│   │   │   ├── piety.lua
│   │   │   ├── playerlist.lua
│   │   │   ├── rainbow.lua
│   │   │   ├── regional.lua
│   │   │   ├── seasons.lua
│   │   │   ├── tag.lua
│   │   │   ├── tOredumonde.lua
│   │   │   └── votekick.lua
│   │   ├── oarc_config.lua
│   │   ├── oarc_events.lua
│   │   ├── permissions.lua
│   │   ├── README.md
│   │   ├── rpg_beastmaster.lua
│   │   ├── rpg_builder.lua
│   │   ├── rpgdata.lua
│   │   ├── rpg.lua
│   │   ├── rpg_permissions.lua
│   │   ├── rpg_pocket_crafter.lua
│   │   ├── rpg_stats.lua
│   │   ├── separate_spawns_guis.lua
│   │   ├── separate_spawns.lua
│   │   ├── survival.lua
│   │   ├── trusted.lua
│   │   ├── utils
│   │   │   ├── antigrief.lua
│   │   │   ├── bot.lua
│   │   │   ├── core.lua
│   │   │   ├── game.lua
│   │   │   ├── gravestone.lua
│   │   │   ├── modular_admin
│   │   │   │   ├── modular_admin_alert.lua
│   │   │   │   ├── modular_admin_boost.lua
│   │   │   │   ├── modular_admin_compensate.lua
│   │   │   │   ├── modular_admin_ghosts.lua
│   │   │   │   ├── modular_admin.lua
│   │   │   │   ├── modular_admin_players.lua
│   │   │   │   ├── modular_admin_spectate.lua
│   │   │   │   └── modular_admin_tag.lua
│   │   │   ├── modular_information
│   │   │   │   ├── modular_information_about.lua
│   │   │   │   ├── modular_information_dummy.lua
│   │   │   │   ├── modular_information.lua
│   │   │   │   ├── modular_information_popup.lua
│   │   │   │   ├── modular_information_rules.lua
│   │   │   │   ├── modular_information_scenario.lua
│   │   │   │   ├── modular_information_stats.lua
│   │   │   │   └── modular_information_team.lua
│   │   │   ├── patreon.lua
│   │   │   ├── perlin.lua
│   │   │   └── topgui.lua
│   │   └── wg_jungle.lua
│   ├── OARC
│   │   ├── blueprint.dat
│   │   ├── config.lua
│   │   ├── control.lua
│   │   ├── info.json
│   │   ├── LICENSE
│   │   ├── locale
│   │   │   ├── frontier_silo.lua
│   │   │   ├── game_opts.lua
│   │   │   ├── LICENSE
│   │   │   ├── oarc_utils.lua
│   │   │   ├── rso
│   │   │   │   ├── metaball.lua
│   │   │   │   ├── prototype_utils.lua
│   │   │   │   ├── rso_config.lua
│   │   │   │   ├── rso_control.lua
│   │   │   │   └── rso_resource_config.lua
│   │   │   ├── tag.lua
│   │   │   └── temp
│   │   │       ├── helper_commands.lua
│   │   │       └── rgcommand.lua
│   │   ├── README.md
│   │   ├── regrowth_map.lua
│   │   ├── separate_spawns_guis.lua
│   │   └── separate_spawns.lua
│   ├── Railworld
│   │   ├── control.lua
│   │   └── locale
│   │       ├── af
│   │       │   └── freeplay.cfg
│   │       ├── ar
│   │       │   └── freeplay.cfg
│   │       ├── be
│   │       │   └── freeplay.cfg
│   │       ├── bg
│   │       │   └── freeplay.cfg
│   │       ├── ca
│   │       │   └── freeplay.cfg
│   │       ├── cs
│   │       │   └── freeplay.cfg
│   │       ├── da
│   │       │   └── freeplay.cfg
│   │       ├── de
│   │       │   └── freeplay.cfg
│   │       ├── el
│   │       │   └── freeplay.cfg
│   │       ├── en
│   │       │   └── freeplay.cfg
│   │       ├── eo
│   │       │   └── freeplay.cfg
│   │       ├── es-ES
│   │       │   └── freeplay.cfg
│   │       ├── et
│   │       │   └── freeplay.cfg
│   │       ├── fi
│   │       │   └── freeplay.cfg
│   │       ├── fr
│   │       │   └── freeplay.cfg
│   │       ├── he
│   │       │   └── freeplay.cfg
│   │       ├── hu
│   │       │   └── freeplay.cfg
│   │       ├── it
│   │       │   └── freeplay.cfg
│   │       ├── ja
│   │       │   └── freeplay.cfg
│   │       ├── ko
│   │       │   └── freeplay.cfg
│   │       ├── lt
│   │       │   └── freeplay.cfg
│   │       ├── lv
│   │       │   └── freeplay.cfg
│   │       ├── nl
│   │       │   └── freeplay.cfg
│   │       ├── no
│   │       │   └── freeplay.cfg
│   │       ├── pl
│   │       │   └── freeplay.cfg
│   │       ├── pt-BR
│   │       │   └── freeplay.cfg
│   │       ├── pt-PT
│   │       │   └── freeplay.cfg
│   │       ├── ro
│   │       │   └── freeplay.cfg
│   │       ├── ru
│   │       │   └── freeplay.cfg
│   │       ├── sk
│   │       │   └── freeplay.cfg
│   │       ├── sl
│   │       │   └── freeplay.cfg
│   │       ├── sr
│   │       │   └── freeplay.cfg
│   │       ├── sv-SE
│   │       │   └── freeplay.cfg
│   │       ├── th
│   │       │   └── freeplay.cfg
│   │       ├── tr
│   │       │   └── freeplay.cfg
│   │       ├── uk
│   │       │   └── freeplay.cfg
│   │       ├── vi
│   │       │   └── freeplay.cfg
│   │       ├── zh-CN
│   │       │   └── freeplay.cfg
│   │       └── zh-TW
│   │           └── freeplay.cfg
│   └── Wasteland
│       ├── antigrief_admin_panel.lua
│       ├── antigrief.lua
│       ├── blueprint.dat
│       ├── bot.lua
│       ├── chatbot.lua
│       ├── control.lua
│       ├── group.lua
│       ├── info.json
│       ├── maps
│       │   ├── cave_miner_changelog.txt
│       │   ├── cave_miner_kaboomsticks.lua
│       │   ├── cave_miner.lua
│       │   ├── cave_miner_market_items.lua
│       │   ├── modules
│       │   │   ├── blueprint_helper.lua
│       │   │   ├── railgun_enhancer.lua
│       │   │   └── teleporters.lua
│       │   └── tools
│       │       ├── cheat_mode.lua
│       │       ├── lazy_chunk_loader.lua
│       │       ├── map_functions.lua
│       │       ├── map_intro.lua
│       │       └── map_pregen.lua
│       ├── player_list.lua
│       ├── poll.lua
│       ├── replay.dat
│       ├── rewards.lua
│       ├── score.lua
│       ├── session_data.lua
│       ├── session_tracker.lua
│       ├── session_updater.lua
│       └── utils
│           ├── corpse_util.lua
│           ├── event.lua
│           ├── game.lua
│           ├── global.lua
│           ├── global_token.lua
│           ├── gui.lua
│           ├── list_utils.lua
│           ├── math.lua
│           ├── perlin_noise.lua
│           ├── PriorityQueue.lua
│           ├── Queue.lua
│           ├── simplex_noise.lua
│           ├── Task.lua
│           └── utils.lua
└── script-output

77 directories, 332 files
```

**Comment by bplein** [comment]
If I run the "find" from the CLI (outside docker at base OS), I get the following output:

```
bill@mothra:~/docker/factorio/factorio1/volume$ cd saves
bill@mothra:~/docker/factorio/factorio1/volume/saves$ SAVES=.
bill@mothra:~/docker/factorio/factorio1/volume/saves$ find -L "$SAVES" -iname \*.zip -mindepth 1 -print | grep -q .
find: warning: you have specified the -mindepth option after a non-option argument -iname, but options are not positional (-mindepth affects tests specified before it as well as those specified after it).  Please specify options before other arguments.

bill@mothra:~/docker/factorio/factorio1/volume/saves$ echo $SAVES
.
bill@mothra:~/docker/factorio/factorio1/volume/saves$ find -L . -iname \*.zip -mindepth 1 -print | grep -q .
find: warning: you have specified the -mindepth option after a non-option argument -iname, but options are not positional (-mindepth affects tests specified before it as well as those specified after it).  Please specify options before other arguments.
```

**Comment by TrueOsiris** [comment]
Same issue with docker on unraid, since the switch to the new docker repo. (0.17.37).
In docker logs I see

`
+ su-exec factorio /opt/factorio/bin/x64/factorio --create /factorio/saves/_autosave1.zip --map-gen-settings /factorio/config/map-gen-settings.json --map-settings /factorio/config/map-settings.json
   0.000 2019-05-07 17:49:11; Factorio 0.17.37 (build 44853, linux64, headless)
`
before it starts the actual
`+ exec su-exec factorio /opt/factorio/bin/x64/factorio --port 34199 --start-server-load-latest --server-settings /factorio/config/server-settings.json --server-banlist /factorio/config/server-banlist.json --rcon-port 27017 --server-whitelist /factorio/config/server-whitelist.json --use-server-whitelist --server-adminlist /factorio/config/server-adminlist.json --rcon-password oy4Ahph9tie4ahc --server-id /factorio/config/server-id.json
`

**Comment by TrueOsiris** [comment]
ok just fixed it by removing the 
´´´
| grep .
´´´ 
in your find lines ...
check it out in my fork:
https://github.com/TrueOsiris/docker_factorio_server/blob/master/0.17/files/docker-entrypoint.sh

**Comment by Macil** [comment]
Isn't that going to always cause that if-block to be skipped? I think find always returns a 0 exit status code. The `| grep -q .` part made it so the combined `find ... | grep -q .` command would fail if nothing was found.

**Comment by Macil** [comment]
@TrueOsiris What error are you getting? Is it during build or during starting the container? If you're getting an error, that sounds different than OP's issue.
I don't think the host OS should make a difference with Docker.

**Comment by Macil** [comment]
I think I found the (OP's) issue. I don't think anything is going wrong in the find commands. In docker-entrypoint.sh, in the line `rm -f "$SAVES/*.tmp.zip"`, only `$SAVES` should be quoted, not the `*` asterisk. Quoting an asterisk causes it to no longer match filenames, and causes it to be treated as a literal asterisk. The line should be `rm -f "$SAVES"/*.tmp.zip`.

**Comment by bplein** [comment]
It’s not just the tmp file section but the save file section as well. And I believe map settings json file which I’ve had problems with disappearing and affecting old maps created by scenarios under 0.16.x that are still being played after upgrading to 0.17.36. Each of my games had scenario-specific settings wiped out causing errors on load.

**Comment by Macil** [comment]
@bplein Can you post the log of the docker image starting up when one of these issues happen, so we can see what code paths docker-entrypoint.sh is triggering?

**Comment by Fank** [comment]
@TrueOsiris can you pleaee create a pull request for this?

**Comment by bplein** [comment]
I’m on a business trip so not so easy to do today. Also I’ve worked around all of these problems by editing and re-editing files and renaming zips etc. so all three containers are running at the moment. I can try and reproduce later tonight and if I am able I will post the Docker logs for each reproduction.

**Comment by bplein** [comment]
I used my VPN and macOS screen sharing to get some old samples that were open in iTerm2 on my desktop :)

```
+ usermod -o -u 845 factorio
usermod: no changes
+ groupmod -o -g 845 factorio
+ chown -R factorio:factorio /factorio
+ SU_EXEC='su-exec factorio'
+ find -L /factorio/saves -iname '*.zip' -mindepth 1 -print
+ grep -q .
+ su-exec factorio /opt/factorio/bin/x64/factorio --create /factorio/saves/_autosave1.zip --map-gen-settings /factorio/config/map-gen-settings.json --map-settings /factorio/config/map-settings.json
   0.000 2019-05-02 18:49:27; Factorio 0.17.35 (build 44769, linux64, headless)
   0.000 Operating system: Linux
   0.000 Program arguments: "/opt/factorio/bin/x64/factorio" "--create" "/factorio/saves/_autosave1.zip" "--map-gen-settings" "/factorio/config/map-gen-settings.json" "--map-settings" "/factorio/config/map-settings.json"
   0.000 Read data path: /opt/factorio/data
```

**Comment by TrueOsiris** [comment]
Created pull request https://github.com/dtandersen/docker_factorio_server/pull/252

**Comment by bplein** [comment]
My users reported 2 of my three servers having problems. One is offline and the other rebooted with a new map. 

Even though we believe this is solved, I’ll capture lots when I get home in a few hours and post them here.

**Comment by bplein** [comment]
I'm guessing that it's not in 0.17.37 image tag yet?
Got home, my first server was running a new game. So I touched the last good save to update the timestamp, docker stop, docker rm, then reran it, and it created a fresh map again.
```
# docker logs -f factorio1
+ set -euo pipefail
+ id
uid=0(root) gid=0(root) groups=0(root),1(bin),2(daemon),3(sys),4(adm),6(disk),10(wheel),11(floppy),20(dialout),26(tape),27(video)
+ FACTORIO_VOL=/factorio
+ mkdir -p /factorio
+ mkdir -p /factorio/saves
+ mkdir -p /factorio/config
+ mkdir -p /factorio/mods
+ mkdir -p /factorio/scenarios
+ mkdir -p /factorio/script-output
+ '[' '!' -f /factorio/config/rconpw ]
+ '[' '!' -f /factorio/config/server-settings.json ]
+ '[' '!' -f /factorio/config/map-gen-settings.json ]
+ '[' '!' -f /factorio/config/map-settings.json ]
+ grep -q .
+ find -L /factorio/saves -iname '*.tmp.zip' -mindepth 1 -print
+ id -u
+ '[' 0 '=' 0 ]
+ usermod -o -u 845 factorio
usermod: no changes
+ groupmod -o -g 845 factorio
+ chown -R factorio:factorio /factorio
+ SU_EXEC='su-exec factorio'
+ grep -q .
+ find -L /factorio/saves -iname '*.zip' -mindepth 1 -print
+ su-exec factorio /opt/factorio/bin/x64/factorio --create /factorio/saves/_autosave1.zip --map-gen-settings /factorio/config/map-gen-settings.json --map-settings /factorio/config/map-settings.json
   0.000 2019-05-09 19:43:22; Factorio 0.17.37 (build 44853, linux64, headless)
   0.000 Operating system: Linux
   0.000 Program arguments: "/opt/factorio/bin/x64/factorio" "--create" "/factorio/saves/_autosave1.zip" "--map-gen-settings" "/factorio/config/map-gen-settings.json" "--map-settings" "/factorio/config/map-settings.json"
   0.000 Read data path: /opt/factorio/data
   0.000 Write data path: /opt/factorio [362350/479189MB]
   0.000 Binaries path: /opt/factorio/bin
```

**Comment by bplein** [comment]
I made my own private image using the docker-entrypoint.sh from @TrueOsiris and it works great. Thanks man!

**Comment by TrueOsiris** [comment]
yw. I updated the originating fork in the new pull request I made. removed the old pull request.
Sorry for all that clumsiness. I'm not a dev.

**Comment by Fank** [comment]
i pushed `factoriotools/factorio:latest` with the latest changes from #253 to fix this issue, please try and tell us if this issue can be closed.

**Comment by bplein** [comment]
with 0.17.38, tmp files are being identified, but not deleted. See below.
```

+ '[' '!' -f /factorio/config/server-settings.json ]
+ '[' '!' -f /factorio/config/map-gen-settings.json ]
+ '[' '!' -f /factorio/config/map-settings.json ]
+ wc -l
+ find -L /factorio/saves -iname '*.tmp.zip' -mindepth 1
+ NRTMPSAVES=1
+ '[' 1 -gt 0 ]
+ rm -f '/factorio/saves/*.tmp.zip'
+ id -u
+ '[' 0 '=' 0 ]
+ usermod -o -u 845 factorio
usermod: no changes
+ groupmod -o -g 845 factorio
+ chown -R factorio:factorio /factorio
+ SU_EXEC='su-exec factorio'
+ wc -l
+ find -L /factorio/saves -iname '*.zip' -mindepth 1
+ NRSAVES=97
+ '[' 97 -eq 0 ]
+ cat /factorio/config/rconpw
+ exec su-exec factorio /opt/factorio/bin/x64/factorio --port 34197 --start-server-load-latest --server-settings /factorio/config/server-settings.json --server-banlist /factorio/config/server-banlist.json --rcon-port 27015 --server-whitelist /factorio/config/server-whitelist.json --use-server-whitelist --server-adminlist /factorio/config/server-adminlist.json --rcon-password eik2ahG7cee3Xee --server-id /factorio/config/server-id.json
   0.000 2019-05-10 16:41:38; Factorio 0.17.38 (build 44914, linux64, headless)
   0.000 Operating system: Linux
   0.000 Program arguments: "/opt/factorio/bin/x64/factorio" "--port" "34197" "--start-server-load-latest" "--server-settings" "/factorio/config/server-settings.json" "--server-banlist" "/factorio/config/server-banlist.json" "--rcon-port" "27015" "--server-whitelist" "/factorio/config/server-whitelist.json" "--use-server-whitelist" "--server-adminlist" "/factorio/config/server-adminlist.json" "--rcon-password" <private> "--server-id" "/factorio/config/server-id.json"
   0.000 Read data path: /opt/factorio/data
   0.000 Write data path: /opt/factorio [361624/479189MB]
   0.000 Binaries path: /opt/factorio/bin
   0.009 System info: [CPU: Intel(R) Xeon(R) CPU           W3670  @ 3.20GHz, 12 cores, RAM: 48284 MB]
   0.009 Environment: DISPLAY=<unset> WAYLAND_DISPLAY=<unset> DESKTOP_SESSION=<unset> XDG_SESSION_DESKTOP=<unset> XDG_CURRENT_DESKTOP=<unset>
   0.009 Running in headless mode
   0.012 Loading mod core 0.0.0 (data.lua)
   0.059 Loading mod base 0.17.38 (data.lua)
   0.274 Loading mod base 0.17.38 (data-updates.lua)
   0.369 Checksum for core: 300852956
   0.369 Checksum of base: 2048447865
   0.894 Info PlayerData.cpp:67: Local player-data.json available, timestamp 1557506496
   0.894 Info PlayerData.cpp:74: Cloud player-data.json unavailable
   0.896 Custom inputs active: 0
   0.897 Factorio initialised
   0.899 Info ServerSynchronizer.cpp:30: nextHeartbeatSequenceNumber(0) initialized Synchronizer nextTickClosureTick(0).
   0.899 Info ServerMultiplayerManager.cpp:776: updateTick(4294967295) changing state from(Ready) to(PreparedToHostGame)
   0.899 Info ServerMultiplayerManager.cpp:776: updateTick(4294967295) changing state from(PreparedToHostGame) to(CreatingGame)
   0.924 Loading map /factorio/saves/railworld.tmp.zip: 88450256 bytes.
   0.925 Error ServerMultiplayerManager.cpp:92: MultiplayerManager failed: "Opening zip /factorio/saves/railworld.tmp.zip failed: Bad zip file"
   0.925 Info ServerMultiplayerManager.cpp:776: updateTick(4294967295) changing state from(CreatingGame) to(InitializationFailed)
   0.925 Info GlobalContext.cpp:683: Waiting for child processes to exit:
   0.926 Info ServerMultiplayerManager.cpp:138: Quitting multiplayer connection.
   0.926 Info ServerMultiplayerManager.cpp:776: updateTick(4294967295) changing state from(InitializationFailed) to(Closed)
   0.944 Info UDPSocket.cpp:210: Closing socket
   0.945 Goodbye
```

**Comment by TrueOsiris** [comment]
@bplein This is a different bug, that was already there imho.
quotes are causing it. I'll make a fix.

This issue can be closed.
